src/mac_features/ack_generator/nrf_802154_ack_data.c

```diff
@@ -321,6 +322,103 @@ static bool addr_index_find(const uint8_t * p_addr,
     return addr_binary_search(p_addr, p_addr_array, p_location, data_type, extended);
 }
 
+/**
+ * @brief Thread implementation of address matching algorithm.
+ *
+ * @param[in]  p_frame  Pointer to the frame for which the ACK frame is being prepared.
+ *
+ * @retval true   Pending bit is to be set.
+ * @retval false  Pending bit is to be cleared.
+ */
+bool addr_match_thread(const uint8_t * p_frame)
+{
+    bool            extended;
+    uint32_t        location;
+    const uint8_t * p_src_addr = nrf_802154_frame_parser_src_addr_get(p_frame, &extended);
+
+    // The pending bit is set by default.
+    if (!m_pending_bit.enabled || NULL == p_src_addr)
+    {
+        return true;
+    }
+
+    return addr_index_find(p_src_addr, &location, NRF_802154_ACK_DATA_PENDING_BIT, extended);
+}
+
+/**
+ * @brief ZigBee implementation of address matching algorithm.
+ *
+ * @param[in]  p_frame  Pointer to the frame for which the ACK frame is being prepared.
+ *
+ * @retval true   Pending bit is to be set.
+ * @retval false  Pending bit is to be cleared.
+ */
+bool addr_match_zigbee(const uint8_t * p_frame)
+{
+    bool ret = false;
+
+    uint8_t                            frame_type;
+    uint8_t                            command_type;
+    nrf_802154_frame_parser_mhr_data_t mhr_fields;
+    const uint8_t                    * p_cmd = p_frame;
+    uint32_t                           location;
+
+    // Check the frame type.
+    frame_type = (p_frame[FRAME_TYPE_OFFSET] & FRAME_TYPE_MASK);
+
+    // Parse the MAC header and retrieve the command type.
+    if (nrf_802154_frame_parser_mhr_parse(p_frame, &mhr_fields))
+    {
+        // Note: Security header is not included in the offset.
+        // If security is to be used at any point, additional calculation
+        // here or in nrf_802154_frame_parser_mhr_parse needs to be implemented.
+        p_cmd += mhr_fields.addressing_end_offset;
+
+        command_type = *p_cmd;
+    }
+    else
+    {
+        // If invalid source or destination addressing mode is detected, assume unknown device.
+        // Command type cannot be checked, as addressing_end_offset value will be invalid.
+        return true;
+    }
+
+    // Check frame type and command type.
+    if (frame_type == FRAME_TYPE_COMMAND && command_type == MAC_CMD_DATA_REQ)
+    {
+        ret = true;
```

Comment:
I think it would be easier to understand if this assignment was placed in an `else` block below.